File: web-lakeville-engineering-hub/src/data/hub-content.ts
/**
 * Static content model for the Engineering Hub: the six categories, their
 * sections, and seed data for tracking screens.
 *
 * Web links start unconfigured so staff can attach the real destinations from
 * inside the app (Settings → Paste a list of links), exactly as on iOS.
 */

export type ResourceActionKind = "web" | "phone" | "email" | "route";

export interface ResourceAction {
  kind: ResourceActionKind;
  /** Default web address, phone number, email, or in-app route. */
  value?: string;
}

/** A single actionable row inside a hub section. */
export interface ResourceLink {
  id: string;
  title: string;
  /** Lucide icon name rendered by `ResourceIcon`. */
  icon: string;
  detail?: string;
  action: ResourceAction;
}

export interface ResourceSection {
  id: string;
  title: string;
  footnote?: string;
  links: ResourceLink[];
}

export interface HubCategory {
  id: string;
  title: string;
  summary: string;
  icon: string;
  sections: ResourceSection[];
}

const DAKOTA_COUNTY_PHONE = "9528917115";
const DAKOTA_COUNTY_EMAIL = "rosalee.mccready@co.dakota.mn.us";

const web = (defaultURL?: string): ResourceAction => ({ kind: "web", value: defaultURL });
const phone = (number: string): ResourceAction => ({ kind: "phone", value: number });
const email = (address: string): ResourceAction => ({ kind: "email", value: address });
const route = (path: string): ResourceAction => ({ kind: "route", value: path });

export const ROW_PERMITTING: HubCategory = {
  id: "row-permitting",
  title: "ROW & Utility Permitting",
  summary:
    "Utility and ROW permit review and coordination: permits, fiber construction, restoration and closeout.",
  icon: "FileSearch",
  sections: [
    {
      id: "row-tracking",
      title: "Permit Tracking",
      footnote: "Always check tracking first to confirm current status before responding.",
      links: [
        {
          id: "row-tracking-all",
          title: "Permit Tracking (ALL Inboxes)",
          icon: "Table2",
          detail: "ROW · RowM · ENG Survey",
          action: route("/permits"),
        },
        { id: "row-applications", title: "ROW Permit Applications", icon: "Inbox", action: web() },
        {
          id: "row-fiber-inquiry",
          title: "ROW/Fiber Inquiry Management",
          icon: "Network",
          action: web(),
        },
        {
          id: "row-moratorium",
          title: "Moratorium Notes & Admin Updates",
          icon: "TriangleAlert",
          action: web(),
        },
      ],
    },
    {
      id: "row-quick-links",
      title: "Quick Links",
      links: [
        { id: "row-onestop-permits", title: "ROW — One Stop Roadway Permits", icon: "Link", action: web() },
        { id: "row-onestop-shop", title: "OneStop Roadway Shop", icon: "Building2", action: web() },
        { id: "row-gis", title: "GIS Lakeville ROW Permits", icon: "Map", action: web() },
        { id: "row-gsoc", title: "FIND One Call Concepts Ticket List", icon: "ListChecks", action: web() },
        { id: "row-contacts", title: "ROW & General Contacts", icon: "Users", action: web() },
        {
          id: "row-dakota-phone",
          title: "Dakota County Permit Office",
          icon: "Phone",
          detail: "(952) 891-7115 — direct applicants here",
          action: phone(DAKOTA_COUNTY_PHONE),
        },
        {
          id: "row-dakota-email",
          title: "Rosalee McCready (Dakota County)",
          icon: "Mail",
          detail: "Internal staff contact",
          action: email(DAKOTA_COUNTY_EMAIL),
        },
      ],
    },
    {
      id: "row-templates",
      title: "Examples & Templates",
      links: [
        {
          id: "row-checklist",
          title: "ROW Checklist",
          icon: "ClipboardCheck",
          detail: "Review steps before approval",
          action: route("/checklists/row"),
        },
        { id: "row-install-example", title: "Installation Plan Example", icon: "FileText", action: web() },
        { id: "row-traffic-example", title: "Traffic Control Plan Example", icon: "TrafficCone", action: web() },
        { id: "row-notice-template", title: "ROW Permit Notice Template", icon: "Files", action: web() },
        { id: "row-fee-permits", title: "2026 Fee Schedule (Permits)", icon: "CircleDollarSign", action: web() },
        { id: "row-fee-full", title: "2026 Fee Schedule (Full)", icon: "CircleDollarSign", action: web() },
      ],
    },
  ],
};

export const SURVEY_REVIEW: HubCategory = {
  id: "survey-review",
  title: "Survey Review",
  summary: "As-built surveys, building permit certificates, checklists and examples.",
  icon: "Ruler",
  sections: [
    {
      id: "survey-inboxes",
      title: "Inboxes",
      links: [
        { id: "survey-rowm", title: "RowM — As-Built Surveys", icon: "Map", action: web() },
        {
          id: "survey-eng",
          title: "ENG Survey — Building Permit Certificates",
          icon: "FileSearch",
          action: web(),
        },
        {
          id: "survey-tracking",
          title: "Permit Tracking (ALL Inboxes)",
          icon: "Table2",
          action: route("/permits"),
        },
      ],
    },
    {
      id: "survey-quick-links",
      title: "Quick Links",
      links: [
        {
          id: "survey-asbuilt-checklist",
          title: "As-Built Survey Checklist",
          icon: "ClipboardCheck",
          action: route("/checklists/as-built"),
        },
        {
          id: "survey-cos-checklist",
          title: "Certificate of Survey Checklist",
          icon: "ClipboardCheck",
          action: route("/checklists/certificate"),
        },
        { id: "survey-county-contacts", title: "Dakota County Surveyor Contacts", icon: "Users", action: web() },
        { id: "survey-gis", title: "GIS Survey Reference Map", icon: "Map", action: web() },
      ],
    },
    {
      id: "survey-templates",
      title: "Examples & Templates",
      links: [
        { id: "survey-asbuilt-example", title: "Approved As-Built Example", icon: "FileText", action: web() },
        { id: "survey-cos-example", title: "Certificate of Survey Example", icon: "FileText", action: web() },
        { id: "survey-comment-template", title: "Review Comment Notice Template", icon: "Files", action: web() },
        { id: "survey-submittal", title: "Survey Submittal Requirements", icon: "ClipboardList", action: web() },
      ],
    },
  ],
};

export const TRAFFIC_RESOURCES: HubCategory = {
  id: "traffic-resources",
  title: "Traffic Resources",
  summary: "Traffic counts, mapping, equipment, data processing and crash-review resources.",
  icon: "CarFront",
  sections: [
    {
      id: "traffic-mapping",
      title: "Mapping & Data",
      links: [
        {
          id: "traffic-count-apps",
          title: "Internal Traffic Count Applications",
          icon: "BarChart3",
          detail: "2026–2029 program",
          action: web(),
        },
        { id: "traffic-count-map", title: "Traffic Count Map (GIS)", icon: "Map", action: web() },
        { id: "traffic-crash-data", title: "Crash Review Data", icon: "TriangleAlert", action: web() },
        { id: "traffic-processing-sop", title: "Data Processing SOP", icon: "Settings2", action: web() },
      ],
    },
    {
      id: "traffic-equipment",
      title: "Equipment & Contacts",
      links: [
        { id: "traffic-equipment-checkout", title: "Count Equipment Checkout", icon: "Package", action: web() },
        { id: "traffic-scheduler", title: "Metro Count Scheduler", icon: "CalendarDays", action: web() },
        { id: "traffic-staff", title: "Traffic Staff Contacts", icon: "Users", action: web() },
      ],
    },
  ],
};

export const CAPITAL_PROJECTS: HubCategory = {
  id: "capital-projects",
  title: "Capital Projects",
  summary: "Quick access to active City project information and coordination resources.",
  icon: "Hammer",
  sections: [
    {
      id: "cip-resources",
      title: "Project Resources",
      links: [
        { id: "cip-map", title: "CIP Project Map (GIS)", icon: "Map", action: web() },
        { id: "cip-notices", title: "Construction Notices & Updates", icon: "Bell", action: web() },
        { id: "cip-detours", title: "Detour & Traffic Impacts", icon: "CarFront", action: web() },
        { id: "cip-contacts", title: "Project Contact List", icon: "Users", action: web() },
      ],
    },
    {
      id: "cip-coordination",
      title: "Coordination",
      links: [
        { id: "cip-utility-meetings", title: "Utility Coordination Meetings", icon: "CalendarDays", action: web() },
        { id: "cip-inspections", title: "Inspection Schedules", icon: "BadgeCheck", action: web() },
        { id: "cip-directory", title: "Consultant & Contractor Directory", icon: "Building2", action: web() },
      ],
    },
  ],
};

export const DEVELOPMENT_REVIEW: HubCategory = {
  id: "development-review",
  title: "Development Review",
  summary: "Plat, site plan and development engineering review resources.",
  icon: "Building2",
  sections: [
    {
      id: "dev-review-types",
      title: "Review Types",
      links: [
        { id: "dev-plat", title: "Plat Review", icon: "Map", action: web() },
        { id: "dev-site-plan", title: "Site Plan Review", icon: "LayoutGrid", action: web() },
        { id: "dev-engineering", title: "Development Engineering Review", icon: "Ruler", action: web() },
      ],
    },
    {
      id: "dev-applications",
      title: "Applications & Tracking",
      links: [
        { id: "dev-tracking", title: "Development Review Tracking", icon: "ClipboardList", action: web() },
        { id: "dev-forms", title: "Application Forms & Fees", icon: "FileClock", action: web() },
        { id: "dev-preapp", title: "Pre-Application Meetings", icon: "Users", action: web() },
        { id: "dev-deadlines", title: "Comment Deadline Calendar", icon: "CalendarDays", action: web() },
      ],
    },
    {
      id: "dev-standards",
      title: "Standards & Templates",
      links: [
        { id: "dev-standards-manual", title: "Design Standards Manual", icon: "BookOpen", action: web() },
        { id: "dev-comment-template", title: "Review Comment Template", icon: "Files", action: web() },
        { id: "dev-plat-checklist", title: "Preliminary Plat Checklist", icon: "ClipboardCheck", action: web() },
        { id: "dev-swppp", title: "Stormwater Requirements (SWPPP)", icon: "CloudRain", action: web() },
      ],
    },
  ],
};

export const PROCEDURES_RESOURCES: HubCategory = {
  id: "procedures-resources",
  title: "Procedures & Resources",
  summary: "Engineering procedures, SOPs, contacts, templates and staff references.",
  icon: "FolderCog",
  sections: [
    {
      id: "proc-sops",
      title: "Procedures & SOPs",
      links: [
        { id: "proc-sop-library", title: "Engineering SOP Library", icon: "BookOpen", action: web() },
        { id: "proc-permit-review", title: "Permit Review Procedure", icon: "ClipboardCheck", action: web() },
        { id: "proc-records", title: "Records Retention & Filing", icon: "Archive", action: web() },
        { id: "proc-safety", title: "Safety & Field Procedures", icon: "ShieldCheck", action: web() },
      ],
    },
    {
      id: "proc-contacts",
      title: "Contacts",
      links: [
        { id: "proc-staff-directory", title: "Engineering Staff Directory", icon: "CircleUser", action: web() },
        {
          id: "proc-dakota-phone",
          title: "Dakota County Permit Office",
          icon: "Phone",
          detail: "(952) 891-7115",
          action: phone(DAKOTA_COUNTY_PHONE),
        },
        { id: "proc-utility-contacts", title: "Utility & Fiber Contacts", icon: "CircleUser", action: web() },
        { id: "proc-gsoc", title: "GSOC / One Call Concepts", icon: "PhoneCall", action: web() },
      ],
    },
    {
      id: "proc-templates",
      title: "Templates & References",
      links: [
        { id: "proc-notice-templates", title: "Notice & Letter Templates", icon: "FileText", action: web() },
        { id: "proc-fees", title: "Fee Schedules (2026)", icon: "CircleDollarSign", action: web() },
        { id: "proc-gis", title: "GIS Maps & Layers", icon: "Map", action: web() },
        { id: "proc-handbook", title: "Employee Handbook References", icon: "BookMarked", action: web() },
      ],
    },
  ],
};

export const HUB_CATEGORIES: HubCategory[] = [
  ROW_PERMITTING,
  SURVEY_REVIEW,
  TRAFFIC_RESOURCES,
  CAPITAL_PROJECTS,
  DEVELOPMENT_REVIEW,
  PROCEDURES_RESOURCES,
];

export function categoryByID(id: string | undefined): HubCategory | null {
  return HUB_CATEGORIES.find((category) => category.id === id) ?? null;
}

export const ALL_LINKS: ResourceLink[] = HUB_CATEGORIES.flatMap((category) =>
  category.sections.flatMap((section) => section.links),
);

export function linkByID(id: string): ResourceLink | null {
  return ALL_LINKS.find((link) => link.id === id) ?? null;
}

/** Which category a link belongs to, used when exporting configuration. */
export function categoryTitleForLink(linkID: string): string {
  for (const category of HUB_CATEGORIES) {
    for (const section of category.sections) {
      if (section.links.some((link) => link.id === linkID)) return category.title;
    }
  }
  return "Other";
}

// MARK: - Checklists

export interface ChecklistDefinition {
  id: string;
  title: string;
  steps: string[];
}

export const CHECKLISTS: ChecklistDefinition[] = [
  {
    id: "row",
    title: "ROW Checklist",
    steps: [
      "Verify moratorium restrictions",
      "Confirm applicable 2026 fees",
      "Review installation plan",
      "Review traffic control plan",
      "Notify fiber contacts",
      "Issue approval notice from template",
    ],
  },
  {
    id: "as-built",
    title: "As-Built Survey Checklist",
    steps: [
      "Confirm submittal is a signed as-built",
      "Check elevations against approved grading plan",
      "Verify utility service locations",
      "Note erosion control status",
      "Log result in tracking spreadsheet",
    ],
  },
  {
    id: "certificate",
    title: "Certificate of Survey Checklist",
    steps: [
      "Confirm building permit number matches",
      "Verify setbacks and lot lines",
      "Check drainage & utility easements",
      "Confirm proposed elevations",
      "Return comments using notice template",
    ],
  },
];

export function checklistByID(id: string | undefined): ChecklistDefinition | null {
  return CHECKLISTS.find((checklist) => checklist.id === id) ?? null;
}

// MARK: - Permit tracking

export const PERMIT_INBOXES = ["row", "rowM", "engSurvey"] as const;
export type PermitInbox = (typeof PERMIT_INBOXES)[number];

export const PERMIT_INBOX_TITLE: Record<PermitInbox, string> = {
  row: "ROW",
  rowM: "RowM — As-Built",
  engSurvey: "ENG Survey",
};

export const PERMIT_STATUSES = [
  "underReview",
  "awaitingApplicant",
  "awaitingRestoration",
  "pendingPickup",
  "approved",
  "closed",
] as const;
export type PermitStatus = (typeof PERMIT_STATUSES)[number];

export const PERMIT_STATUS_TITLE: Record<PermitStatus, string> = {
  underReview: "Under review",
  awaitingApplicant: "Awaiting applicant",
  awaitingRestoration: "Awaiting restoration",
  pendingPickup: "Pending pickup",
  approved: "Approved",
  closed: "Closed",
};

export const OPEN_PERMIT_STATUSES: PermitStatus[] = [
  "underReview",
  "awaitingApplicant",
  "awaitingRestoration",
  "pendingPickup",
];

export interface PermitRecord {
  id: string;
  number: string;
  applicant: string;
  inbox: PermitInbox;
  status: PermitStatus;
  receivedDate: string;
  note?: string;
}

function daysAgo(days: number): string {
  const date = new Date();
  date.setDate(date.getDate() - days);
  return date.toISOString();
}

/** Sample rows so the tracking board isn't empty on first run. */
export function seedPermits(): PermitRecord[] {
  return [
    { id: "seed-1", number: "ROW-2026-0141", applicant: "Xcel Energy", inbox: "row", status: "awaitingRestoration", receivedDate: daysAgo(6), note: "Restoration due before winter moratorium." },
    { id: "seed-2", number: "ROW-2026-0139", applicant: "Frontier Fiber", inbox: "row", status: "underReview", receivedDate: daysAgo(8), note: "Traffic control plan needs revision." },
    { id: "seed-3", number: "ROW-2026-0136", applicant: "Dakota Electric", inbox: "row", status: "approved", receivedDate: daysAgo(14) },
    { id: "seed-4", number: "AB-2026-0087", applicant: "Keavy & Klein Surveying", inbox: "rowM", status: "approved", receivedDate: daysAgo(2) },
    { id: "seed-5", number: "AB-2026-0086", applicant: "Sathre-Bergquist", inbox: "rowM", status: "underReview", receivedDate: daysAgo(3) },
    { id: "seed-6", number: "CS-2026-0032", applicant: "Lennar Homes", inbox: "engSurvey", status: "pendingPickup", receivedDate: daysAgo(1) },
    { id: "seed-7", number: "CS-2026-0031", applicant: "M/I Homes", inbox: "engSurvey", status: "awaitingApplicant", receivedDate: daysAgo(5) },
  ];
}

File: ios-lakeville-engineering-hub/LakevilleEngineeringHub/Data/HubContent.swift
import Foundation

/// Static content model for the Engineering Hub: the six categories, their
/// sections, and seed data for tracking screens. Web links start unconfigured
/// so staff can attach the real SharePoint URLs from inside the app.
nonisolated enum HubContent {
    static let dakotaCountyPhone = "9528917115"
    static let dakotaCountyContactEmail = "rosalee.mccready@co.dakota.mn.us"

    static let categories: [HubCategory] = [
        rowPermitting,
        surveyReview,
        trafficResources,
        capitalProjects,
        developmentReview,
        proceduresResources,
    ]

    static func category(id: String) -> HubCategory? {
        categories.first { $0.id == id }
    }

    // MARK: - ROW & Utility Permitting

    static let rowPermitting = HubCategory(
        id: "row-permitting",
        title: "ROW & Utility Permitting",
        summary: "Permit tracking, utility coordination, fiber construction, restoration and closeout.",
        headerSummary: "Utility and ROW permit review and coordination: permits, fiber construction, restoration and closeout.",
        systemImage: "doc.text.magnifyingglass",
        sections: [
            ResourceSection(
                id: "row-tracking",
                title: "Permit Tracking",
                footnote: "Always check tracking first to confirm current status before responding.",
                links: [
                    ResourceLink(
                        id: "row-tracking-all",
                        title: "Permit Tracking (ALL Inboxes)",
                        systemImage: "tablecells",
                        detail: "ROW · RowM · ENG Survey",
                        action: .route(.permitTracking)
                    ),
                    ResourceLink(id: "row-applications", title: "ROW Permit Applications", systemImage: "tray.full"),
                    ResourceLink(id: "row-fiber-inquiry", title: "ROW/Fiber Inquiry Management", systemImage: "point.3.connected.trianglepath.dotted"),
                    ResourceLink(id: "row-moratorium", title: "Moratorium Notes & Admin Updates", systemImage: "exclamationmark.triangle"),
                ]
            ),
            ResourceSection(
                id: "row-quick-links",
                title: "Quick Links",
                links: [
                    ResourceLink(id: "row-onestop-permits", title: "ROW — One Stop Roadway Permits", systemImage: "link"),
                    ResourceLink(id: "row-onestop-shop", title: "OneStop Roadway Shop", systemImage: "building.columns"),
                    ResourceLink(id: "row-gis", title: "GIS Lakeville ROW Permits", systemImage: "map"),
                    ResourceLink(id: "row-gsoc", title: "FIND One Call Concepts Ticket List", systemImage: "list.bullet.rectangle"),
                    ResourceLink(id: "row-contacts", title: "ROW & General Contacts", systemImage: "person.2"),
                    ResourceLink(
                        id: "row-dakota-phone",
                        title: "Dakota County Permit Office",
                        systemImage: "phone",
                        detail: "(952) 891-7115 — direct applicants here",
                        action: .phone(dakotaCountyPhone)
                    ),
                    ResourceLink(
                        id: "row-dakota-email",
                        title: "Rosalee McCready (Dakota County)",
                        systemImage: "envelope",
                        detail: "Internal staff contact",
                        action: .email(dakotaCountyContactEmail)
                    ),
                ]
            ),
            ResourceSection(
                id: "row-templates",
                title: "Examples & Templates",
                links: [
                    ResourceLink(
                        id: "row-checklist",
                        title: "ROW Checklist",
                        systemImage: "checklist",
                        detail: "Review steps before approval"
                    ),
                    ResourceLink(id: "row-install-example", title: "Installation Plan Example", systemImage: "doc.richtext"),
                    ResourceLink(id: "row-traffic-example", title: "Traffic Control Plan Example", systemImage: "cone"),
                    ResourceLink(id: "row-notice-template", title: "ROW Permit Notice Template", systemImage: "doc.on.doc"),
                    ResourceLink(id: "row-fee-permits", title: "2026 Fee Schedule (Permits)", systemImage: "dollarsign.circle"),
                    ResourceLink(id: "row-fee-full", title: "2026 Fee Schedule (Full)", systemImage: "dollarsign.circle"),
                ]
            ),
        ]
    )

    // MARK: - Survey Review

    static let surveyReview = HubCategory(
        id: "survey-review",
        title: "Survey Review",
        summary: "As-built surveys, building permit certificates, checklists and examples.",
        headerSummary: "As-built surveys, building permit certificates, checklists and examples.",
        systemImage: "camera.metering.matrix",
        sections: [
            ResourceSection(
                id: "survey-inboxes",
                title: "Inboxes",
                links: [
                    ResourceLink(id: "survey-rowm", title: "RowM — As-Built Surveys", systemImage: "map"),
                    ResourceLink(id: "survey-eng", title: "ENG Survey — Building Permit Certificates", systemImage: "doc.text.magnifyingglass"),
                    ResourceLink(
                        id: "survey-tracking",
                        title: "Permit Tracking (ALL Inboxes)",
                        systemImage: "tablecells",
                        action: .route(.permitTracking)
                    ),
                ]
            ),
            ResourceSection(
                id: "survey-quick-links",
                title: "Quick Links",
                links: [
                    ResourceLink(id: "survey-asbuilt-checklist", title: "As-Built Survey Checklist", systemImage: "checklist"),
                    ResourceLink(id: "survey-cos-checklist", title: "Certificate of Survey Checklist", systemImage: "checklist"),
                    ResourceLink(id: "survey-county-contacts", title: "Dakota County Surveyor Contacts", systemImage: "person.2"),
                    ResourceLink(id: "survey-gis", title: "GIS Survey Reference Map", systemImage: "map"),
                ]
            ),
            ResourceSection(
                id: "survey-templates",
                title: "Examples & Templates",
                links: [
                    ResourceLink(id: "survey-asbuilt-example", title: "Approved As-Built Example", systemImage: "doc.richtext"),
                    ResourceLink(id: "survey-cos-example", title: "Certificate of Survey Example", systemImage: "doc.richtext"),
                    ResourceLink(id: "survey-comment-template", title: "Review Comment Notice Template", systemImage: "doc.on.doc"),
                    ResourceLink(id: "survey-submittal", title: "Survey Submittal Requirements", systemImage: "list.clipboard"),
                ]
            ),
        ]
    )

    // MARK: - Traffic Resources

    static let trafficResources = HubCategory(
        id: "traffic-resources",
        title: "Traffic Resources",
        summary: "Traffic counts, mapping, equipment, data processing and crash-review resources.",
        headerSummary: "Traffic counts, mapping, equipment, data processing and crash-review resources.",
        systemImage: "car.2",
        sections: [
            ResourceSection(
                id: "traffic-mapping",
                title: "Mapping & Data",
                links: [
                    ResourceLink(
                        id: "traffic-count-apps",
                        title: "Internal Traffic Count Applications",
                        systemImage: "chart.bar.doc.horizontal",
                        detail: "2026–2029 program"
                    ),
                    ResourceLink(id: "traffic-count-map", title: "Traffic Count Map (GIS)", systemImage: "map"),
                    ResourceLink(id: "traffic-crash-data", title: "Crash Review Data", systemImage: "exclamationmark.triangle"),
                    ResourceLink(id: "traffic-processing-sop", title: "Data Processing SOP", systemImage: "gearshape.2"),
                ]
            ),
            ResourceSection(
                id: "traffic-equipment",
                title: "Equipment & Contacts",
                links: [
                    ResourceLink(id: "traffic-equipment-checkout", title: "Count Equipment Checkout", systemImage: "shippingbox"),
                    ResourceLink(id: "traffic-scheduler", title: "Metro Count Scheduler", systemImage: "calendar"),
                    ResourceLink(id: "traffic-staff", title: "Traffic Staff Contacts", systemImage: "person.2"),
                ]
            ),
        ]
    )

    // MARK: - Capital Projects

    static let capitalProjects = HubCategory(
        id: "capital-projects",
        title: "Capital Projects",
        summary: "Quick access to active City project information and coordination resources.",
        headerSummary: "Quick access to active City project information and coordination resources.",
        systemImage: "hammer",
        sections: [
            ResourceSection(
                id: "cip-resources",
                title: "Project Resources",
                links: [
                    ResourceLink(id: "cip-map", title: "CIP Project Map (GIS)", systemImage: "map"),
                    ResourceLink(id: "cip-notices", title: "Construction Notices & Updates", systemImage: "bell"),
                    ResourceLink(id: "cip-detours", title: "Detour & Traffic Impacts", systemImage: "car"),
                    ResourceLink(id: "cip-contacts", title: "Project Contact List", systemImage: "person.2"),
                ]
            ),
            ResourceSection(
                id: "cip-coordination",
                title: "Coordination",
                links: [
                    ResourceLink(id: "cip-utility-meetings", title: "Utility Coordination Meetings", systemImage: "calendar"),
                    ResourceLink(id: "cip-inspections", title: "Inspection Schedules", systemImage: "checkmark.seal"),
                    ResourceLink(id: "cip-directory", title: "Consultant & Contractor Directory", systemImage: "building.2"),
                ]
            ),
        ]
    )

    // MARK: - Development Review

    static let developmentReview = HubCategory(
        id: "development-review",
        title: "Development Review",
        summary: "Plat, site plan and development engineering review resources.",
        headerSummary: "Plat, site plan and development engineering review resources.",
        systemImage: "building.2",
        sections: [
            ResourceSection(
                id: "dev-review-types",
                title: "Review Types",
                links: [
                    ResourceLink(id: "dev-plat", title: "Plat Review", systemImage: "map"),
                    ResourceLink(id: "dev-site-plan", title: "Site Plan Review", systemImage: "square.grid.3x3.square"),
                    ResourceLink(id: "dev-engineering", title: "Development Engineering Review", systemImage: "camera.metering.matrix"),
                ]
            ),
            ResourceSection(
                id: "dev-applications",
                title: "Applications & Tracking",
                links: [
                    ResourceLink(id: "dev-tracking", title: "Development Review Tracking", systemImage: "list.clipboard"),
                    ResourceLink(id: "dev-forms", title: "Application Forms & Fees", systemImage: "doc.badge.clock"),
                    ResourceLink(id: "dev-preapp", title: "Pre-Application Meetings", systemImage: "person.3"),
                    ResourceLink(id: "dev-deadlines", title: "Comment Deadline Calendar", systemImage: "calendar"),
                ]
            ),
            ResourceSection(
                id: "dev-standards",
                title: "Standards & Templates",
                links: [
                    ResourceLink(id: "dev-standards-manual", title: "Design Standards Manual", systemImage: "book"),
                    ResourceLink(id: "dev-comment-template", title: "Review Comment Template", systemImage: "doc.on.doc"),
                    ResourceLink(id: "dev-plat-checklist", title: "Preliminary Plat Checklist", systemImage: "checklist"),
                    ResourceLink(id: "dev-swppp", title: "Stormwater Requirements (SWPPP)", systemImage: "cloud.rain"),
                ]
            ),
        ]
    )

    // MARK: - Procedures & Resources

    static let proceduresResources = HubCategory(
        id: "procedures-resources",
        title: "Procedures & Resources",
        summary: "Engineering procedures, SOPs, contacts, templates and staff references.",
        headerSummary: "Engineering procedures, SOPs, contacts, templates and staff references.",
        systemImage: "folder.badge.gearshape",
        sections: [
            ResourceSection(
                id: "proc-sops",
                title: "Procedures & SOPs",
                links: [
                    ResourceLink(id: "proc-sop-library", title: "Engineering SOP Library", systemImage: "book"),
                    ResourceLink(id: "proc-permit-review", title: "Permit Review Procedure", systemImage: "checklist"),
                    ResourceLink(id: "proc-records", title: "Records Retention & Filing", systemImage: "archivebox"),
                    ResourceLink(id: "proc-safety", title: "Safety & Field Procedures", systemImage: "shield.lefthalf.filled"),
                ]
            ),
            ResourceSection(
                id: "proc-contacts",
                title: "Contacts",
                links: [
                    ResourceLink(id: "proc-staff-directory", title: "Engineering Staff Directory", systemImage: "person.crop.circle"),
                    ResourceLink(
                        id: "proc-dakota-phone",
                        title: "Dakota County Permit Office",
                        systemImage: "phone",
                        detail: "(952) 891-7115",
                        action: .phone(dakotaCountyPhone)
                    ),
                    ResourceLink(id: "proc-utility-contacts", title: "Utility & Fiber Contacts", systemImage: "person.crop.circle"),
                    ResourceLink(id: "proc-gsoc", title: "GSOC / One Call Concepts", systemImage: "phone.badge.waveform"),
                ]
            ),
            ResourceSection(
                id: "proc-templates",
                title: "Templates & References",
                links: [
                    ResourceLink(id: "proc-notice-templates", title: "Notice & Letter Templates", systemImage: "doc.text"),
                    ResourceLink(id: "proc-fees", title: "Fee Schedules (2026)", systemImage: "dollarsign.circle"),
                    ResourceLink(id: "proc-gis", title: "GIS Maps & Layers", systemImage: "map"),
                    ResourceLink(id: "proc-handbook", title: "Employee Handbook References", systemImage: "book.closed"),
                ]
            ),
        ]
    )

    // MARK: - Checklists

    static let rowChecklistSteps: [String] = [
        "Verify moratorium restrictions",
        "Confirm applicable 2026 fees",
        "Review installation plan",
        "Review traffic control plan",
        "Notify fiber contacts",
        "Issue approval notice from template",
    ]

    static let asBuiltChecklistSteps: [String] = [
        "Confirm submittal is a signed as-built",
        "Check elevations against approved grading plan",
        "Verify utility service locations",
        "Note erosion control status",
        "Log result in tracking spreadsheet",
    ]

    static let certificateChecklistSteps: [String] = [
        "Confirm building permit number matches",
        "Verify setbacks and lot lines",
        "Check drainage & utility easements",
        "Confirm proposed elevations",
        "Return comments using notice template",
    ]

    // MARK: - Seed tracking data

    static func seedPermits(referenceDate: Date = Date()) -> [PermitRecord] {
        let calendar = Calendar.current
        func daysAgo(_ days: Int) -> Date {
            calendar.date(byAdding: .day, value: -days, to: referenceDate) ?? referenceDate
        }
        return [
            PermitRecord(number: "ROW-2026-0141", applicant: "Xcel Energy", inbox: .row, status: PermitStatus.awaitingRestoration.rawValue, receivedDate: daysAgo(6), note: "Restoration due before winter moratorium."),
            PermitRecord(number: "ROW-2026-0139", applicant: "Frontier Fiber", inbox: .row, status: PermitStatus.underReview.rawValue, receivedDate: daysAgo(8), note: "Traffic control plan needs revision."),
            PermitRecord(number: "ROW-2026-0136", applicant: "Dakota Electric", inbox: .row, status: PermitStatus.approved.rawValue, receivedDate: daysAgo(14)),
            PermitRecord(number: "AB-2026-0087", applicant: "Keavy & Klein Surveying", inbox: .rowM, status: PermitStatus.approved.rawValue, receivedDate: daysAgo(2)),
            PermitRecord(number: "AB-2026-0086", applicant: "Sathre-Bergquist", inbox: .rowM, status: PermitStatus.underReview.rawValue, receivedDate: daysAgo(3)),
            PermitRecord(number: "CS-2026-0032", applicant: "Lennar Homes", inbox: .engSurvey, status: PermitStatus.pendingPickup.rawValue, receivedDate: daysAgo(1)),
            PermitRecord(number: "CS-2026-0031", applicant: "M/I Homes", inbox: .engSurvey, status: PermitStatus.awaitingApplicant.rawValue, receivedDate: daysAgo(5)),
        ]
    }

    static let projects: [CapitalProject] = [
        CapitalProject(
            id: "185th-street",
            name: "185th St Reconstruction — Phase 2",
            phase: "Construction",
            progress: 0.45,
            manager: "City Engineering",
            contractor: "Northwest Asphalt",
            schedule: "May 2026 – Oct 2026",
            scope: "Full-depth reconstruction, storm sewer replacement, trail and signal improvements between Kenwood Trail and Ipava Ave."
        ),
        CapitalProject(
            id: "kenwood-trail",
            name: "Kenwood Trail Improvements",
            phase: "Design",
            progress: 0.20,
            manager: "City Engineering",
            contractor: "Consultant design underway",
            schedule: "Design 2026 · Construction 2027",
            scope: "Pedestrian crossing upgrades, trail widening and drainage improvements along Kenwood Trail."
        ),
        CapitalProject(
            id: "mill-overlay-2026",
            name: "2026 Mill & Overlay Program",
            phase: "Construction",
            progress: 0.60,
            manager: "City Engineering",
            contractor: "Bituminous Roadways",
            schedule: "Jun 2026 – Sep 2026",
            scope: "Annual pavement management program: milling, paving, casting adjustments and striping across selected neighborhoods."
        ),
        CapitalProject(
            id: "holyoke-water-main",
            name: "Holyoke Ave Water Main",
            phase: "Utility Coordination",
            progress: 0.15,
            manager: "City Engineering / Utilities",
            contractor: "TBD — bidding fall 2026",
            schedule: "Coordination 2026 · Construction 2027",
            scope: "Water main replacement with private utility relocation coordination through the downtown corridor."
        ),
    ]

    static let trafficCounts: [TrafficCount] = [
        TrafficCount(id: "tc-1", location: "210th St W", averageDailyTraffic: 8_600, month: "Apr", isLatest: false),
        TrafficCount(id: "tc-2", location: "Dodd Blvd", averageDailyTraffic: 15_200, month: "May", isLatest: false),
        TrafficCount(id: "tc-3", location: "233rd St W", averageDailyTraffic: 12_400, month: "Jun", isLatest: false),
        TrafficCount(id: "tc-4", location: "Kenwood Trl", averageDailyTraffic: 18_900, month: "Jul", isLatest: false),
        TrafficCount(id: "tc-5", location: "Keats Ave", averageDailyTraffic: 4_100, month: "Aug", isLatest: true),
    ]
}
